corehq/apps/es/management/commands/ensure_indices_reindexed.py

```diff
@@ -0,0 +1,165 @@
+from django.core.management.base import BaseCommand, CommandError
+
+from corehq.apps.es import CANONICAL_NAME_ADAPTER_MAP, const
+from corehq.apps.es.client import manager
+from corehq.apps.es.transient_util import doc_adapter_from_cname
+
+helptext = """
+The management command is intended for execution during Django migrations and is designed for use
+in Elasticsearch upgrades. Its purpose is to ensure the reindexing of all Elasticsearch indexes.
+The command accomplishes this by comparing the document counts in both primary and secondary indexes for each app.
+If the difference in count is not zero for indexes or exceeds 1% for high-frequency indices such as
+case-search, cases, and forms, the command will fail.
+However, it will skip execution if the current Elasticsearch version is greater than the specified version
+which means that Elasticsearch is already upgraded.
+"""
+
+
+class Command(BaseCommand):
+
+    help = helptext
+
+    REASONABLE_DIFF_PERCENT = 1
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--target_es_version', type=int, choices=[1, 2, 5], required=True,
```

Comment:
[Required](https://docs.python.org/3/library/argparse.html#required) arguments conventionally do not have the `--` option prefix.
```suggestion
            'target_es_version', type=int, choices=[1, 2, 5],
```

Reply:
Thanks. I did not know this 💡

Reply:
https://github.com/dimagi/commcare-hq/pull/33949/commits/06a6faec3a45de5c931967e8c9f962f516dee728